http/rest-client-reactive/src/main/java/io/quarkus/ts/http/restclient/reactive/resources/SseResource.java

```diff
@@ -0,0 +1,67 @@
+package io.quarkus.ts.http.restclient.reactive.resources;
+
+import java.util.Arrays;
+import java.util.concurrent.locks.LockSupport;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.client.ClientBuilder;
+import jakarta.ws.rs.client.WebTarget;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.sse.OutboundSseEvent;
+import jakarta.ws.rs.sse.Sse;
+import jakarta.ws.rs.sse.SseEventSink;
+import jakarta.ws.rs.sse.SseEventSource;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+
+/**
+ * Reproducer resource for https://github.com/quarkusio/quarkus/issues/36986
+ */
+@ApplicationScoped
+@Path("sse")
+public class SseResource {
+    @Context
+    Sse sse;
+
+    @GET
+    @Path("client")
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response clientUpdate() {
+        String host = ConfigProvider.getConfig().getValue("quarkus.http.host", String.class);
+        int port = ConfigProvider.getConfig().getValue("quarkus.http.port", Integer.class);
+        StringBuilder eventCapture = new StringBuilder();
+
+        WebTarget target = ClientBuilder.newClient().target("http://" + host + ":" + port + "/sse/server-update");
```

Comment:
It seems, that the whole clientUpdate can be moved to the test class and it will be greatly simplified in such case. Am I wrong?

Reply:
AFAIK you cannot do that. The whole point of SSE testing is, that server is asking itself  via HTTP.